lib/bfp.ts

```diff
@@ -69,7 +69,7 @@ export class Bfp {
         signProgressCallback?: Function,
         signFinishedCallback?: Function,
         uploadProgressCallback?: Function,
-        uploadFinishedCallback?: Function) {
+        uploadFinishedCallback?: Function): Promise<string> {
```

Comment:
I don't think we should use a promise here, or enforce return type of a string.  The callbacks should fire immediately when they're called

Reply:
Hello,

This doesn't return a promise. All `async` functions in Typescript have return types `Promise<something>`.

This line meant to say "This function doesn't return anything" so I had to change this or the newer TypeScript compiler wouldn't compile your regtest test, saying "This function return void, you can't take its return value".

If you want to unenforce all checks, then the correct way to do that is to add `: any` so that the compiler knows that it normally returns *something*.

Reply:
my bad, I misread the line.